src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/AssetCacheEntry.cs

```diff
@@ -1,11 +1,75 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+using System;
+using UnityEngine;
+
 namespace Microsoft.MixedReality.SpectatorView
 {
+    [Serializable]
+    internal class AssetId
+    {
+        public static AssetId Empty { get; } = new AssetId(System.Guid.Empty, -1);
+
+        public StringGuid Guid => guid;
+
+        [SerializeField]
+        private StringGuid guid;
+
+        public long FileIdentifier => fileIdentifier;
+
+        [SerializeField]
+        private long fileIdentifier;
+
+        public AssetId(StringGuid guid, long fileIdentifier)
+        {
+            this.guid = guid;
+            this.fileIdentifier = fileIdentifier;
+        }
+
+        public override bool Equals(object obj)
+        {
+            AssetId assetId = obj as AssetId;
+            if (assetId == null)
+            {
+                return false;
+            }
+
+            return this == assetId;
+        }
+
+        public override int GetHashCode()
+        {
+            return FileIdentifier.GetHashCode() ^ Guid.ToString().GetHashCode();

```

Comment:
>return FileIdentifier.GetHashCode() ^ Guid.ToString().GetHashCode(); [](start = 12, length = 68)

[https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=dotnet-plat-ext-3.0](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=dotnet-plat-ext-3.0)

Reply:
going to keep this hash code function for now based on non-deterministic scenarios for that combine/dependency on .net extensions.